backends/s3/s3.go

```diff
@@ -189,9 +183,39 @@ func (b *Backend) Store(ctx context.Context, name string, data []byte) error {
 		return err
 	}
 	if b.opt.UseUpdateMarker {
+		b.mu.Lock()
```

Comment:
Again, this locks around a network request, blocking other goroutines for potentially a long time.

Also note that a defer is not scoped to this block. This is fine now in this case, because it occurs at the end of the function, but can cause trouble if another network operation is added at the end of this function later.